code/obj/augmentations.dm

```diff
@@ -0,0 +1,357 @@
+			return
+		aholder.removeAbility(abil)
+
+	///What should happen every life cycle while an aug is broken
+	proc/on_broken(var/mult = 1)
+		if (broken)
+			return TRUE
+		return FALSE
+
+	disposing()
+		src.owner = null
+		src.owner_organ = null
+		..()
+
+ABSTRACT_TYPE(/obj/item/augmentation/head)
+/obj/item/augmentation/head //second abstract parent incase other augmentation types get added
+	name = "surgical augmentation parent"
+	icon_state = "augmentation"
+
+/obj/item/augmentation/head/wireless_interact //you can interact with mechanical things at range at the cost of flash vulnerability
+	name = "wireless interactor"
+	icon_state = "augmentation_wire"
+	desc = "An augmentation that allows for ranged interaction with various electronic devices."
+
+	proc/ranged_click(mob/user, atom/target, location, control)
+		var/mob/M = src.owner
+		if(src.hasStatus("flashed"))
+			return
+		if(get_dist(user, target) <= 1)
+			return
+		if (M.client.check_any_key(KEY_EXAMINE | KEY_POINT) || ishelpermouse(target) || M.equipped()) // slightly hacky, oh well, tries to check whether we want to click normally or use attack_ai
+			return
+		else
+			if (get_dist(M, target) > 0)
+				set_dir(get_dir(M, target))
+			if(istype(target, /obj/machinery))
+				var/turf/target_turf = get_turf(target)
+				var/obj/overlay/energy = new/obj/overlay(target_turf)
+				energy.icon = 'icons/effects/effects.dmi'
+				energy.icon_state = "energytwirlin_fast"
+				energy.name = "electronic energy pulse"
+				energy.anchored = 1
+				SPAWN_DBG(57 CENTI SECONDS)
+					if (energy)
+						qdel(energy)
+			target.attack_ai(M)
+
+	proc/flash_check()
+		src.setStatus("flashed", duration = 15 SECONDS)
+		src.take_damage(5, 5, 0) //owie
+		src.owner.remove_stamina(25)
+		if(src.broken)
+			src.owner.remove_stamina(15)
+
+	on_organ_transplant(var/mob/M as mob)
+		..()
+		if(!broken)
+			RegisterSignal(src.owner, COMSIG_LIVING_CLICK, .proc/ranged_click)
+			M.mob_flags |= USR_DIALOG_UPDATES_RANGE
+		RegisterSignal(src.owner, COMSIG_MOB_FLASHED, .proc/flash_check)
+
+	on_organ_removal()
+		..()
+		var/mob/M = src.owner
+		if(!broken)
+			UnregisterSignal(src.owner, COMSIG_LIVING_CLICK)
+			M.mob_flags &= ~USR_DIALOG_UPDATES_RANGE
+		UnregisterSignal(src.owner, COMSIG_MOB_FLASHED)
+
+	on_broken(var/mult = 1)
+		if (!..())
+			return
+		src.owner.reagents.add_reagent("nanites", 0.5 * mult) //you want borg powers? Well, come and get 'em!
+
+	breakme()
+		..()
+		var/mob/M = src.owner
+		if(M.mob_flags & USR_DIALOG_UPDATES_RANGE)
+			UnregisterSignal(src.owner, COMSIG_LIVING_CLICK)
+			M.mob_flags &= ~USR_DIALOG_UPDATES_RANGE
+
+	organ_is_valid(var/obj/item/organ/chosen_organ)
+		if(istype(chosen_organ, /obj/item/organ/brain) && !chosen_organ.robotic && !chosen_organ.synthetic)
+			return TRUE
+		else
+			return FALSE
+
+
+/datum/abilityHolder/augmentation //unused currently but laying the path for future augmentations
+	usesPoints = 0
+	regenRate = 0
+	tabName = "Body"
```

Comment:
not a big fan of including unused code, there should be at least on example for people to copy and base their work on.